src/TestProblem/Hydro/Riemann/Init_TestProb_Hydro_Riemann.cpp

```diff
@@ -425,7 +425,11 @@ void SetGridIC( real fluid[], const double x, const double y, const double z, co
                                     EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table );
 
 // do NOT include magnetic energy here
-   fluid[ENGY] = Hydro_ConEint2Etot( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], Eint, 0.0 );
+   fluid[ENGY   ] = Hydro_ConEint2Etot( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], Eint, 0.0 );
+#  ifdef TEMP_IG
+   fluid[TEMP_IG] = EoS_DensEint2Temp_CPUPtr( fluid[DENS], fluid[ENGY], fluid+NCOMP_FLUID,
```

Comment:
`fluid[TEMP_IG]` is not initialized before invoking `EoS_DensEint2Temp_CPUPtr()`, and so for the initial guess of temperature.